File: unibakonto/src/main/java/sk/pluk64/unibakonto/TransactionItem.java
package sk.pluk64.unibakonto;

import org.jsoup.nodes.Element;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Iterator;

public class TransactionItem {
    public final String timestamp;
    private transient Date parsedTimestamp;
    public final String service;
    public final String shortcut;
    public final String description;
    public final String amount;
    public final double parsedAmount;
    public final String method;
    public final String obj;
    public final String payed;

    private static final SimpleDateFormat dateFormat = new SimpleDateFormat("d. MM. yyyy HH:mm:ss");

    public TransactionItem(Element tableRow) {
        Iterator<Element> columns = tableRow.children().iterator();

        timestamp = columns.next().text();
        service = columns.next().text();
        shortcut = columns.next().text();
        description = columns.next().text();
        amount = columns.next().text();
        method = columns.next().text();
        obj = columns.next().text();
        payed = columns.next().text();

        parsedAmount = Double.parseDouble(amount.replace(',', '.'));
    }

    public Date getParsedTimestamp() {
        if (parsedTimestamp == null) {
            try {
                parsedTimestamp = dateFormat.parse(timestamp);
            } catch (ParseException e) {
                e.printStackTrace();
                return Calendar.getInstance().getTime();
            }
        }
        return parsedTimestamp;
    }
}

File: unibakonto/src/main/java/sk/pluk64/unibakonto/TransactionItem.kt
package sk.pluk64.unibakonto

import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import org.jsoup.nodes.Element
import java.text.ParseException
import java.text.SimpleDateFormat

@Serializable
data class TransactionItem(
    val timestamp: String = "",
    val service: String = "",
    val shortcut: String = "",
    val description: String = "",
    val amount: String = "",
    val method: String = "",
    val obj: String = "",
    val payed: String = ""
) {
    @Transient
    val parsedTimestamp by lazy {
        try {
            DATE_FORMAT.parse(timestamp)
        } catch (e: ParseException) {
            e.printStackTrace()
            null
        }
    }

    @Transient
    val parsedAmount by lazy {
        try {
            amount.replace(',', '.').toDouble()
        } catch (e: NumberFormatException) {
            e.printStackTrace()
            null
        }
    }

    companion object {
        fun fromElement(tableRow: Element): TransactionItem {
            val columns = tableRow.children().iterator()

            val timestamp = columns.next().text()
            val service = columns.next().text()
            val shortcut = columns.next().text()
            val description = columns.next().text()
            val amount = columns.next().text()
            val method = columns.next().text()
            val obj = columns.next().text()
            val payed = columns.next().text()

            return TransactionItem(timestamp, service, shortcut, description, amount, method, obj, payed)
        }

        private val DATE_FORMAT
            get() = SimpleDateFormat("d. MM. yyyy HH:mm:ss")
    }
}
